ast/src/parsed/mod.rs

```diff
@@ -638,6 +640,78 @@ pub enum BinaryOperator {
     Greater,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum BinaryOperatorAssociativity {
+    Left,
+    Right,
+    RequireParentheses,
+}
+
+trait Precedence {
+    fn precedence(&self) -> ExpressionPrecedence;
+}
+
+impl Precedence for UnaryOperator {
+    fn precedence(&self) -> ExpressionPrecedence {
+        use UnaryOperator::*;
+        match self {
+            // NOTE: Any modification must be done with care to not overlap with BinaryOperator's precedence
+            Next => 1,
+            Minus | LogicalNot => 2,
+        }
+    }
+}
+
+impl Precedence for BinaryOperator {
+    fn precedence(&self) -> ExpressionPrecedence {
+        use BinaryOperator::*;
+        match self {
+            // NOTE: Any modification must be done with care to not overlap with LambdaExpression's precedence
+            // Unary Oprators
+            // **
+            Pow => 3,
+            // * / %
+            Mul | Div | Mod => 4,
+            // + -
+            Add | Sub => 5,
+            // << >>
+            ShiftLeft | ShiftRight => 6,
+            // &
+            BinaryAnd => 7,
+            // ^
+            BinaryXor => 8,
+            // |
+            BinaryOr => 9,
+            // = += -= *= /= %= &= |= ^= <<= >>=
```

Comment:
This is not the right comment. Identity is `=`. Please also combine this with the next line